src/arith/rewrite_simplify.cc

```diff
@@ -353,6 +353,19 @@ PrimExpr RewriteSimplifier::Impl::VisitExpr_(const SubNode* op) {
         truncdiv(x + c1, c3) - truncdiv(x, c3), truncdiv(truncmod(x, c3) + c1, c3),
         CanProveGreaterEqual(x.Eval(), 0) && c1.Eval()->value >= 0 && c3.Eval()->value > 0);
 
+    TVM_TRY_REWRITE_IF(truncdiv(x * c1 + c2, c3) - truncdiv(x * c1 + c4, c3), truncdiv(c2 - c4, c3),
+                       c1.Eval()->value >= 0 && c2.Eval()->value > 0 && c3.Eval()->value > 0 &&
+                           c4.Eval()->value >= 0 && CanProveGreaterEqual(x.Eval(), 0));
```

Comment:
hmm,  I am wondering if this always holds.  

say `x = 0, c1 = 1, c2 = 3, c3 = 3, c4 = 1`
`truncdiv(x * c1 + c2, c3) - truncdiv(x * c1 + c4, c3) = = truncdiv(3, 3) - truncdiv(1, 3)`
`truncdiv(c2 - c4, c3) = truncdiv(2, 3)`

Reply:
Yes, need to add condition: c2-c4>=c3